spharpy/spherical.py

```diff
@@ -1,104 +1,585 @@
+     It can be 'pseudo_inverse', 'quadrature', or None.
+        - ``'pseudo_inverse'``: Uses the Moore-Penrose pseudo-inverse
+         for the inverse transform.
+        - ``'quadrature'``: Uses quadrature for the inverse transform.
+        - ``None``: No inverse transform is applied.
+    """
+
+    def __init__(
+        self,
+        n_max,
+        coords,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_transform=None,
+        weights=None,
+        condon_shortley=True
+    ):
+        r"""
+        Initialize the SphericalHarmonics class.
+
+        Parameters
+        ----------
+        n_max : int
+            Maximum spherical harmonic order
+        coords : :doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>`
+            Coordinate object with sampling points for
+            which the basis matrix is calculated
+        basis_type : str, optional
+            Type of spherical harmonic basis, either ``'complex'`` or
+            ``'real'``. The default is ``'complex'``.
+        normalization : str, optional
+            Normalization convention, either ``'n3d'``, ``'maxN'`` or
+             ``'sn3d'``. The default is ``'n3d'``.
+            (maxN is only supported up to 3rd order)
+        channel_convention : str, optional
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        inverse_transform : str, optional
+            Inverse transform type, either ``'pseudo_inverse'`` or
+            ``'quadrature'``. The default is None.
+        weights : array-like, optional
+            Sampling weights for the quadrature transform.
+            Required if `quadrature` is chosen.
+        condon_shortley : bool, optional
+            Whether to include the Condon-Shortley phase term.
+            The default is True.
+        """
+        if not isinstance(coords, pf.Coordinates):
+            raise TypeError("coords must be a pyfar.Coordinates object")
+        self.n_max = n_max
+        self.weights = weights
+        self.coords = coords
+        self.basis_type = basis_type
+        self.inverse_transform = inverse_transform
+        self.channel_convention = channel_convention  # ch. ord. conv.
+        self.normalization = normalization  # gain norm. conv.
+        self.basis = None
+        (self.basis_gradient_theta,
+         self.basis_gradient_phi) = (None, None)
+        self._basis_inv = None
+        (self._basis_inv_gradient_theta,
+         self._basis_inv_gradient_phi) = (None, None)
+        self._condon_shortley = condon_shortley
+        # Store previous values for comparison
+        self._prev_n_max = self.n_max
+        self._prev_coords = self.coords
+        self._prev_basis_type = self.basis_type
+        self._prev_inverse_transform = self.inverse_transform
```

Comment:
This can lead to memory explosion for large sampling grids (I.e. HRTF or directivity data interpolation via SH)
Is this really necessary?